### DISCUSSION TITLE
How to output flight data like running script in UE?

### DISCUSSION THREAD

**Discussion by Yuhandswash** [JSBSim-Team/jsbsim#901] [Support requests]
We can set the <output> properties in the aircraft.xml.
for example, in c172x.xml.
```xml
    <output name="JSBout172B.csv" type="CSV" rate="10">
        <rates> ON             </rates>
        <velocities> ON        </velocities>
        <position> ON          </position>
        <atmosphere> OFF       </atmosphere>
        <fcs> ON               </fcs>
        <ground_reactions> OFF </ground_reactions>
        <propulsion> ON       </propulsion>
        <simulation> ON        </simulation>
        <massprops> ON        </massprops>
        <forces> OFF           </forces>
        <moments> OFF          </moments>
        <aerosurfaces> OFF      </aerosurfaces>
        <property> position/vrp-gc-latitude_deg </property>
        <property> position/vrp-longitude_deg   </property>
        <property> position/vrp-radius-ft       </property>
        <function name="velocities/pi-deg_sec">
          <todegrees> <p> velocities/pi-rad_sec </p> </todegrees>
        </function>
    </output>
```
Run script c1723.xml in JSBSim,You can get the output flight data file JSBout172B.csv.

If I add <output> properties in the 787-7.xml in UEReferenceApp. Can I get flight data by running UEReferenceApp project?

**Comment by Yuhandswash** [comment]
If I add <output> properties in the 787-7.xml in UEReferenceApp. Can I get flight data by running UEReferenceApp project?

**Comment by gallonmate** [comment]
Yes you just need to specify a location like this example:
```xml
    <output name="C:\JSBSimLogs\JSBout172C.csv" type="CSV" rate="60">
```

**Reply by Yuhandswash** [reply]
Thank you!!!

**Reply by Yuhandswash** [reply]
but it doesn't work. I add output properties in 787-8.xml like this,  
`    <output name="D:\JSBSimLogs\JSBout787-8.csv" type="CSV" rate="60">
		<simulation> OFF </simulation>
		<atmosphere> OFF </atmosphere>
		<masspros> OFF </masspros>
		<aerosurfaces> OFF </aerosurfaces>
		<rate> ON </rate>
		<velocities> ON </velocities>
		<forces> ON </forces>
		<moments> ON </moments>
 	    <position> ON </position>
		<propulsion> OFF </propulsion>
		<fcs> OFF </fcs>
		<ground_reactions> OFF </ground_reactions>
		<coefficients> OFF </coefficients>
    </output>`
The path to 787-8.xml is "D:\UE5 Projects\jsbsim-master 2023.2.22\jsbsim-master\aircraft\787-8". 
I ran the UEReferenceApp.uproject. The path to UEReferenceApp.uproject is D:\UE5 Projects\jsbsim-master 2023.2.22\jsbsim-master\UnrealEngine. (In this uproject, Aircraft Model of JSBSimMovement is 787-8.)
As a result, no flight data appears under the JSBSimLogs folder

**Reply by gallonmate** [reply]
Unreal Plugin when compiling makes an entire copy of JSBSim repo folder and has it's own separate folder for aircraft. It only does this once when you first build the plugin. So editing aircraft later has no effect. (I know it's confusing lol)

You have to edit the aircraft in the Unreal Project folder. 
The UE5 aircrafts for you would be located: `D:\UE5 Projects\jsbsim-master 2023.2.22\jsbsim-master\UnrealEngine\Plugins\JSBSimFlightDynamicsModel\Resources\JSBSim\aircraft`

Once you edit and save the aircraft in that folder it should work in UE5.

**Reply by Yuhandswash** [reply]
Thank you very much. I didn't notice that before
